src/format.rs

```diff
@@ -222,16 +231,20 @@ async fn run(
         };
 
         let start_row = &buf[(ctx.codeblock_start as i32 + offset) as usize];
-        let whitespace = utils::get_start_whitespace(start_row);
+        let codeblock_indent = utils::get_start_whitespace(start_row);
+
+        let content_indent =
+            utils::get_start_whitespace(&buf[(ctx.start as i32 + offset) as usize]);
 
         let mut fixed_output = String::new();
         for line in output.lines() {
-            fixed_output.push_str(&whitespace);
+            fixed_output.push_str(&codeblock_indent);
+            fixed_output.push_str(&content_indent);
```

Comment:
Reading this again now, this is not correct. The content indent already includes the codeblock indent. It's just using the start row, not the column.

The easiest solution would be to just use content indent and ignore codeblock indent. Then it will just keep the indent of the first conent line.
I guess the other, more elaborate solution would be to take the codeblock indent, and have a map from codeblock indent to content indent per language. This way the indent will always get fixed. So markdown and org would be 0, restructuredtext would be 4, etc.

Actually, I think for now the easy solution is good enough.

Reply:
Right, my test examples did not have an indented codeblock, so I hadn't noticed.

Fixed it now.